packages/grafana-ui/src/components/IconButton/IconButton.tsx

```diff
@@ -46,12 +48,21 @@ export const IconButton = React.forwardRef<HTMLButtonElement, Props>(
     ref
   ) => {
     const theme = useTheme2();
-    const styles = getStyles(theme, size, variant);
+    let limitedIconSize: LimitedIconSize;
+
+    // very large icons (xl to xxxl) are unified to size xl
+    if (size === 'xxl' || size === 'xxxl') {
+      limitedIconSize = 'xl';
```

Comment:
Probably it would make sense to log a deprecation warning in this case, e.g.:

```typescript
import { deprecationWarning } from '@grafana/data';

// ...

deprecationWarning('IconButton', 'size="xxl" and size="xxxl"', 'size="xl"');

```

Reply:
That's an interesting tool. I didn't know we had something like this. 💡